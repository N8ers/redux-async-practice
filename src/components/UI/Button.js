import styles from "./Button.module.css";

function Button(props) {
  const { color, title, handleClick } = props;

  const clickHandler = () => {
    handleClick();
  };

  return (
    <button className={styles[color]} onClick={clickHandler}>
      {title}
    </button>
  );
}

export default Button;
