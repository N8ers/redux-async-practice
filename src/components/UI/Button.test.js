import { render, screen } from "@testing-library/react";
import Button from "./Button";

describe("Greeting Component", () => {
  test("renders prop content", () => {
    render(<Button color="primary" title="test button" />);

    const button = screen.getByText("test button");
    expect(button).toBeInTheDocument();
  });
  test("renders prop content snap", () => {
    render(<Button color="secondary" title="navigate back" />);

    const button = screen.getByText("navigate back");
    expect(button).toMatchSnapshot();
  });
});
