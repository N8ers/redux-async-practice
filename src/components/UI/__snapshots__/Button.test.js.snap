// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Greeting Component renders prop content snap 1`] = `
<button
  class="secondary"
>
  navigate back
</button>
`;
